File: analise/color_statistics.py
from config.color import Color


class ColorStatistics:
    def __init__(self, color: Color):
        self.stats = None
        self.color = color
        self.clear_stats()

    def add_half(self, bottom: bool, left: bool):
        if bottom:
            self.stats['half_left_bottom' if left else 'half_right_bottom'] += 1
        else:
            self.stats['half_left_up' if left else 'half_right_up'] += 1

    def add_full(self, left: bool):
        self.stats['full_left' if left else 'full_right'] += 1

    def clear_stats(self):
        self.stats = {'full_left': 0,
                      'full_right': 0,
                      'half_left_up': 0,
                      'half_right_up': 0,
                      'half_left_bottom': 0,
                      'half_right_bottom': 0
                      }

    def __str__(self):
        stats = ', '.join(f'{key}={value}' for key, value in self.stats.items())
        return f"{self.color.name}: {stats}"

File: src/main/java/ornaments/classes/ColorStatistics.java
package ornaments.classes;


public class ColorStatistics {
    private final OrnamentColor color;
    private int fullLeft;
    private int fullRight;
    private int halfLeftUp;
    private int halfRightUp;
    private int halfLeftBottom;
    private int halfRightBottom;


    public ColorStatistics(OrnamentColor color) {
        this.color = color;
        clearStats();
    }

    public OrnamentColor getColor() {
        return color;
    }

    public void addHalf(boolean bottom, boolean left) {
        if (bottom) {
            if (left) {
                halfLeftBottom++;
            } else {
                halfRightBottom++;
            }
        } else {
            if (left) {
                halfLeftUp++;
            } else {
                halfRightUp++;
            }
        }
    }

    public void addFull(boolean left) {
        if (left) {
            fullLeft++;
        } else {
            fullRight++;
        }
    }

    public void clearStats() {
        fullLeft = 0;
        fullRight = 0;
        halfLeftUp = 0;
        halfRightUp = 0;
        halfLeftBottom = 0;
        halfRightBottom = 0;
    }

    public String toString() {
        return color.name() + ": " +
                "\n\tfullLeft=" + fullLeft +
                ", fullRight=" + fullRight +
                ", halfLeftUp=" + halfLeftUp +
                ", halfRightUp=" + halfRightUp +
                ", halfLeftBottom=" + halfLeftBottom +
                ", halfRightBottom=" + halfRightBottom + "\n";
    }
}
